File: VB/FilterColumnChooser/ColumnFilterMultiValueConverter.vb
Imports DevExpress.Xpf.Grid
Imports System
Imports System.Collections
Imports System.Globalization
Imports System.Linq
Imports System.Windows.Data

Namespace FilterColumnChooserExample
	Public Class ColumnFilterMultiValueConverter
		Implements IMultiValueConverter

		Public Function Convert(ByVal values() As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IMultiValueConverter.Convert
			If values(0) Is Nothing Then
				Return Nothing
			End If
			Dim columns = DirectCast(values(0), IList).Cast(Of BaseColumn)()
			Dim filter As String = values(1).ToString()
			If String.IsNullOrEmpty(filter) Then
				Return columns
			End If
			Return columns.Where(Function(col) col.HeaderCaption.ToString().StartsWith(filter, True, culture))
		End Function
		Public Function ConvertBack(ByVal value As Object, ByVal targetTypes() As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object() Implements IMultiValueConverter.ConvertBack
			Throw New NotImplementedException()
		End Function
	End Class
End Namespace

File: CS/FilterColumnChooser/ColumnFilterMultiValueConverter.cs
using DevExpress.Xpf.Grid;
using System;
using System.Collections;
using System.Globalization;
using System.Linq;
using System.Windows.Data;

namespace FilterColumnChooserExample
{
    public class ColumnFilterMultiValueConverter : IMultiValueConverter
    {
        public object Convert(object[] values, Type targetType, object parameter, CultureInfo culture) {
            if (values[0] == null) return null;
            var columns = ((IList)values[0]).Cast<BaseColumn>();
            string filter = values[1].ToString();
            if (String.IsNullOrEmpty(filter))
                return columns;
            return columns.Where(col => col.HeaderCaption.ToString().StartsWith(filter, true, culture));
        }
        public object[] ConvertBack(object value, Type[] targetTypes, object parameter, CultureInfo culture) {
            throw new NotImplementedException();
        }
    }
}
